app/service/PdfService.scala

```diff
@@ -59,9 +60,9 @@ class PdfService @Inject() (connector: PdfGeneratorServiceConnector, appConfig:
         None
     }
 
-  def generatePdf(htmlContent: Html): Future[PdfFile] = {
-    val pdf = connector.generatePdf(htmlContent)
-    pdf
+  def generatePdf(htmlContent: Html): Future[Array[Byte]] = {
+    val xlsTransformer = Source.fromResource("view_application.xml").mkString
+    pdfGeneratorService.render(htmlContent, xlsTransformer)
```

Comment:
I think the generatePdf() method can be added to the I think new PdfGeneratorService and then this can be renamed to be something like en EncryptionService or something. From a quick glance where this is injected, once we move the method to the new PdfGeneratorService we only need to inject that then to be able to render pdfs and I feel we are at the moment mixing something which handles Encryption and pdf generation at the moment.

Reply:
I left this intentionally for now as there is a lot of additional code for the retrieval of PDF/HTML which is probably all unnecessary (check the ApplicationController). The encryption code isn't used anymore (nearly certain) and can be cleaned up when any updates which are made to the ruling certificate are transferred to this repository.